chain/pool/src/lib.rs

```diff
@@ -6,16 +6,19 @@ use near_primitives::types::AccountId;
 
 pub use crate::types::Error;
 use near_primitives::hash::CryptoHash;
+use std::cell::Cell;
 
 pub mod types;
 
+type TxMap = HashMap<(AccountId, PublicKey), Vec<SignedTransaction>>;
+
 /// Transaction pool: keeps track of transactions that were not yet accepted into the block chain.
 #[derive(Default)]
 pub struct TransactionPool {
     /// Transactions grouped by a pair of (account ID, signer public key).
     /// It's more efficient to keep transactions unsorted and with potentially conflicting nonce
     /// than create a BTreeMap for every transaction on average.
-    pub transactions: HashMap<(AccountId, PublicKey), Vec<SignedTransaction>>,
+    pub transactions: Cell<TxMap>,
```

Comment:
It does not look like we need to wrap it into a `Cell`. At least it does not seem like we are using features provided by it. 
If the problem is in the compiler not understanding usage of both `unique_transactions` and `transactions` then you can deconstruct `let Self{unique_transactions, transactions } = self;`.